ejb3/src/main/java/org/jboss/as/ejb3/component/EJBValidationConfigurator.java

```diff
@@ -52,27 +50,18 @@ private EJBValidationConfigurator() {
 
     @Override
     public void configure(final DeploymentPhaseContext context, final ComponentDescription description, final ComponentConfiguration configuration) throws DeploymentUnitProcessingException {
-        final DeploymentReflectionIndex deploymentReflectionIndex = context.getDeploymentUnit().getAttachment(Attachments.REFLECTION_INDEX);
-        final ClassReflectionIndex classIndex = deploymentReflectionIndex.getClassIndex(configuration.getComponentClass());
+        final ClassReflectionIndex classIndex = context.getDeploymentUnit().getAttachment(Attachments.REFLECTION_INDEX).getClassIndex(configuration.getComponentClass());
         final Constructor<?> ctor = classIndex.getConstructor(new String[0]);
         boolean noInterface = false;
         for(ViewDescription view : description.getViews()) {
             if(view.getViewClassName().equals(description.getComponentClassName())) {
                 noInterface = true;
             }
         }
-        if(ctor == null && noInterface) {
-            //we only validate this for no interface views
-            throw EjbLogger.ROOT_LOGGER.ejbMustHavePublicDefaultConstructor(description.getComponentName(), description.getComponentClassName());
-        }
-        if(configuration.getComponentClass().getEnclosingClass() != null) {
-            throw EjbLogger.ROOT_LOGGER.ejbMustNotBeInnerClass(description.getComponentName(), description.getComponentClassName());
-        }
-        if(!Modifier.isPublic(configuration.getComponentClass().getModifiers())) {
-            throw EjbLogger.ROOT_LOGGER.ejbMustBePublicClass(description.getComponentName(), description.getComponentClassName());
-        }
-        if(Modifier.isFinal(configuration.getComponentClass().getModifiers())) {
-            throw EjbLogger.ROOT_LOGGER.ejbMustNotBeFinalClass(description.getComponentName(), description.getComponentClassName());
-        }
+
+        EjbValidationsUtil.verifyEjbClassAndDefaultConstructor(ctor, configuration.getComponentClass().getEnclosingClass(), noInterface, description.getComponentName(), description.getComponentClassName(), configuration.getComponentClass().getModifiers());
+        EjbValidationsUtil.verifyEjbMethodAreNotFinalNorStatic(configuration.getComponentClass().getDeclaredMethods(),description.getComponentClassName());
+        for ( Class<?> interfaceClass : configuration.getComponentClass().getInterfaces())
+            EjbValidationsUtil.verifyEjbMethodAreNotFinalNorStatic(interfaceClass.getDeclaredMethods(), interfaceClass.getCanonicalName());
```

Comment:
I wasn't sure, but a quick test shows me that neither static nor final are allowed on interfaces

Reply:
@kabir excellent point, I've removed that.